Plugins/Json/include/Acts/Plugins/Json/GridJsonConverter.hpp

```diff
@@ -66,12 +75,64 @@ nlohmann::json toJson(const grid_type& grid) {
       }
     }
   }
-  jGrid["axes"] = jAxes;
   jGrid["data"] = jData;
 
   return jGrid;
 }
 
+/// @brief Templated grid conversion to json
+///
+/// @tparam grid_type the type of the grid
+/// @param grid the grid object
+///
+/// @return a json object to represent the grid
+template <typename grid_type>
+nlohmann::json toJsonDetray(const grid_type& grid) {
+  nlohmann::json jGrid;
+
+  auto axes = grid.axes();
+  nlohmann::json jAxes;
+
+  // Fill the axes
+  for (unsigned int ia = 0u; ia < grid_type::DIM; ++ia) {
+    auto jAxis = AxisJsonConverter::toJsonDetray(*axes[ia]);
+    jAxis["label"] = ia;
+    jAxes.push_back(jAxis);
+  }
+  jGrid["axes"] = jAxes;
+
+  nlohmann::json jData;
+  // 1D connections
+  if constexpr (grid_type::DIM == 1u) {
+    for (unsigned int ib0 = 1u; ib0 <= axes[0u]->getNBins(); ++ib0) {
+      typename grid_type::index_t lbin;
+      lbin[0u] = ib0;
+      nlohmann::json jBin;
+      jBin["loc_index"] = lbin;
+      jBin["content"] = grid.atLocalBins(lbin);
+      jData.push_back(jBin);
+    }
+  }
+
+  // 2D connections
+  if constexpr (grid_type::DIM == 2u) {
+    for (unsigned int ib0 = 1u; ib0 <= axes[0u]->getNBins(); ++ib0) {
+      for (unsigned int ib1 = 1u; ib1 <= axes[1u]->getNBins(); ++ib1) {
+        typename grid_type::index_t lbin;
+        lbin[0u] = ib0;
+        lbin[1u] = ib1;
+        nlohmann::json jBin;
+        jBin["loc_index"] = lbin;
+        jBin["content"] = grid.atLocalBins(lbin);
+        jData.push_back(jBin);
+      }
+    }
+  }
```

Comment:
```suggestion
typename grid_type::index_t lbin;
for (unsigned int ib0 = 1u; ib0 <= axes[0u]->getNBins(); ++ib0) {
      lbin[0u] = ib0;
      // 2D connections
      if constexpr (grid_type::DIM == 2u) {
          for (unsigned int ib1 = 1u; ib1 <= axes[1u]->getNBins(); ++ib1) {
              lbin[1u] = ib1;
          }
       }
   }
  nlohmann::json jBin;
  jBin["loc_index"] = lbin;
  jBin["content"] = grid.atLocalBins(lbin);
  jData.push_back(jBin);
```
Might be easier to extend to 3D grids at some point

Reply:
That suggestion doesn't work, the filling of the content has to be once within the tighter, once without the outer loop.  I could do a fake inner loop with dim=1 and then with an if not fill `bin[1u]` but that doesn't make it clearer.